fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -333,6 +335,38 @@ public void setTransactionIsTracedSupplier(Supplier<Boolean> transactionIsTraced
         return transactionIsTracedSupplier;
     }
 
+    /**
+     * Controls if calls to <code>FDBRecordContext.asyncToSync()</code> or <code>FDBDatabase.asyncToAsync()</code>
+     * will attempt to detect when they are being called from within an asynchronous context and how they should
+     * react to this fact, when they are. Note that the process of performing this detection is quite expensive
+     * so running with detection enabled in not recommended for environments other than testing.
+     *
+     * @param behavior the blocking desired blocking detection behavior
+     * (see {@link BlockingInAsyncDetection})
+     */
+    public void setBlockingInAsyncDetection(BlockingInAsyncDetection behavior) {
+        setBlockingInAsyncDetection(() -> behavior);
+    }
+
+    /**
+     * Provides a supplier tha controls if calls to <code>FDBRecordContext.asyncToSync()</code>
+     * or <code>FDBDatabase.asyncToAsync()</code> * will attempt to detect when they are being called from
+     * within an asynchronous context and how they should * react to this fact, when they are.  Because such detection
+     * is quite expensive, it is suggested that it is either <code>DISABLED</code> for anything other than testing
+     * environments, or that the supplier randomly chooses a small sample rate in which detection should be
+     * enabled.
```

Comment:
It looks like the formatting of this Javadoc got messed up at some point.

The code references should be links (for the methods at least), and I guess so should `DISABLED` (maybe?).